openSearchLane.go

```diff
@@ -0,0 +1,277 @@
+package lane
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/opensearch-project/opensearch-go/v3"
+	"github.com/opensearch-project/opensearch-go/v3/opensearchapi"
+)
+
+type LogTemplate struct {
+	AppName      string            `json:"appName"`
+	ParentLaneId string            `json:"parentLaneId,omitempty"`
+	JourneyID    string            `json:"journeyId,omitempty"`
+	LaneID       string            `json:"laneId"`
+	LogMessage   string            `json:"logMessage"`
+	Metadata     map[string]string //TODO
+}
+
+type openSearchLane struct {
+	logLane
+	openSearchConnection *openSearchConnection
+	wg                   sync.WaitGroup
+}
+
+type openSearchConnection struct {
```

Comment:
Not finding the go routine; I guess it is still in the todo list.

The go routine will have

```
select {
case ... // the case(s) needed to move messages out of logBuffer and into opensearch

case pwg := <-me.refCountCh:
   me.refs--
   if me.refs == 0 {
      // do the quiesque and close (blocking)
   }
   *pwg.Done()
}
```

Reply:
the goroutine is `flushLogBuffer()`